stac_fastapi/pgstac/stac_fastapi/pgstac/core.py

```diff
@@ -27,6 +27,8 @@
 class CoreCrudClient(AsyncBaseCoreClient):
     """Client for core endpoints defined by stac."""
 
+    search_request_model: Type[PgstacSearch] = attr.ib(init=False, default=PgstacSearch)
```

Comment:
It might be possible to avoid this duplication of `search_request_model` specification if the client methods that use it were higher order functions. This would mean taking the `StacApi` specified `search_request_model` and returning a function which would then be registered as an endpoint.